shared/src/types/keccak.rs

```diff
@@ -103,13 +104,48 @@ pub fn keccak_hash(bytes: &[u8]) -> KeccakHash {
 
 /// This module defines encoding methods compatible with Ethereum
 /// smart contracts.
+// TODO: relocate this sub-module to a new home
```

Comment:
I've been thinking about this for a while too. Maybe we should just have an ABI mod which has both encode and keccak submods